### PR TITLE
[MU4] Rendering midi in parts

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

File: framework/midi/internal/midiportdatasender.h [hunk 1]
```diff
@@ -19,14 +19,17 @@
 #ifndef MU_MIDI_MIDIPORTDATASENDER_H
 #define MU_MIDI_MIDIPORTDATASENDER_H
 
+#include <mutex>
```
Comment (psmokotnin) [inline framework/midi/internal/midiportdatasender.h:22]: not needed here

File: framework/audio/devtools/audioenginedevtools.cpp [hunk 2]
```diff
@@ -99,24 +99,41 @@ void AudioEngineDevTools::makeArpeggio()
...
+    t.channels.push_back(0);
+    m_midiStream->initData.tracks.push_back(t);
+
+    midi::Event e;
+    e.channel = 0;
+    e.type = EventType::ME_PROGRAM;
+    e.a = 0;
+    m_midiStream->initData.initEvents.push_back(e);
+
+    auto makeChunk = [this](Chunk& chunk, uint32_t tick, int pitch) {
```
Comment (psmokotnin) [inline framework/audio/devtools/audioenginedevtools.cpp:113]: void VSTDevTools::makeArpeggio() need to be changed too
Or makeArpeggio can be moved to midi module

File: framework/audio/devtools/audioenginedevtools.cpp [hunk 3]
```diff
@@ -99,24 +99,41 @@ void AudioEngineDevTools::makeArpeggio()
...
-    m_midiStream->request.onReceive(this, [this, makeEvents](uint32_t tick) {
+    Track t;
+    t.num = 0;
+    t.channels.push_back(0);
+    m_midiStream->initData.tracks.push_back(t);
+
+    midi::Event e;
+    e.channel = 0;
+    e.type = EventType::ME_PROGRAM;
+    e.a = 0;
```
Comment (psmokotnin) [inline framework/audio/devtools/audioenginedevtools.cpp:110]: It's better to use `Event(channel_t ch, EventType type, int a, int b)`
Event::a and Event::b goes to be private

File: framework/midi/internal/midiportdatasender.h (hunk 1 above)
Comment (igorkorsukov) [reply framework/midi/internal/midiportdatasender.h]: removed

File: framework/audio/devtools/audioenginedevtools.cpp (hunk 2 above)
Comment (igorkorsukov) [reply framework/audio/devtools/audioenginedevtools.cpp:113]: fixed

**Review by psmokotnin — APPROVED** [review]

**Pull request merged**